aerogear/unifiedpush-operator
# Ups apns token

## Motivation
Implements support for IOSTokenVariant CRD which lets users setup a token based ios variant in UPS.

## What
 * Create iOSTokenVariant CRD, example CR and controller
 * Support Create, Get, and Delete operations for iOSTokenVariants
 * Validation of fields.

## Why
Tokens are an alternative way to communicate with APNS and is a feature the community has asked for.

## How
Originally, we wanted to overload the iOSVariant CR.  This proved to not be vaiable because the validation rules we wanted to set were not supported by code generation.  iOSTokenVariant was created in place of that; in the future we should probably refactor a bunch of this to use more regular patterns.

## Verification Steps
1. Deploy the operator with minishift
2. Deploy a ups and push application with the example crs.
3. Get a teamId, keyId, bundleId, and private key from APNS or from @ziccardi 
4. Edit deploy/crds/examples/push_v1alpha1_iostokenvariant_cr.yaml with those values
5. Confirm that Creating, reading, and deleting the variant records those actions in UPS correctly
6. If you have access to a iOS device you can test sending in UPS, but this has been testing in UPS already.

## Checklist:


## Progress
